src/installation/guides/zfs.md

```diff
@@ -0,0 +1,225 @@
+| -f       | Force the creation of the pool                    |
+| -m none  | Set the mountpoint to none                        |
+| zroot    | The name of the pool                              |
+| dev      | The device to be used in the creation of the pool |
+|          | (ie. ata-XXXXX-partX or wwn-XXXX-partX)           |
+
+To ensure the pool's creation was successful use:
+
+```
+$ zpool list
+NAME    SIZE  ALLOC   FREE  CKPOINT  EXPANDSZ   FRAG    CAP  DEDUP    HEALTH  ALTROOT
+zroot   928G   432K   928G        -         -     0%     0%  1.00x    ONLINE  -
+```
+
+Then create the following data sets, which are ZFS's equivalent to partitions:
+
+```
+# zfs create -o mountpoint=none zroot/ROOT
+# zfs create -o mountpoint=/ zroot/ROOT/void
+cannot mount '/': directory is not empty
+property may be set but unable to remount filesystem
+```
+
+At this point users may additionally want to set additional options to their
+datasets. An explanation of the available options is avaiable under the Native
+Properties section of
+[zfs(8)](https://man.voidlinux.org/zfs.8#Native_Properties).
+
+## Base Installation
+
+To mount the system for installation export then re-import the pool at an
+alternate mountpoint:
+
+```
+# zpool export zroot
+# zpool import -R /mnt/void zroot
+```
+
+Then consult the [chroot guide](./) with the installation of the base system and
+its configuration.
+
+**Important:** After consulting the guide, install `grub` and `zfs` from the new
+system which will build the requirements necessary to boot: `(chroot)#
+xbps-install -S -R $REPO grub zfs`
+
+## GRUB Installation
+
+Confirm the ZFS modules are loaded in the new system using
+[lsmod(8)](https://man.voidlinux.org/lsmod.8):
+
+```
+(chroot)$ lsmod | grep zfs
+zfs                  4022272  1
+zunicode              335872  1 zfs
+zlua                  172032  1 zfs
+zavl                   16384  1 zfs
+icp                   294912  1 zfs
+zcommon                86016  2 zfs,icp
+znvpair                69632  2 zfs,zcommon
+spl                   102400  5 zfs,icp,znvpair,zcommon,zavl
+```
+
+Setup the ZFS cachefile, to ensure the new machine recognizes the previously
+created pool properly.
+
+`(chroot)# zpool set cachefile=/etc/zfs/zpool.cache zroot`
+
+Then notate the bootable system for GRUB's autoconfig:
+
+`(chroot)# zpool set bootfs=zroot/ROOT/void zroot`
+
+Finally ensure GRUB recognizes the ZFS module before being installed.
+
+Note: When using id's the ZPOOL_VDEV_NAME_PATH variable must be set to 1.
+
+```
+(chroot)# ZPOOL_VDEV_NAME_PATH=1 grub-probe /
+zfs
+```
+
+Install GRUB onto the drive specifying the path:
+
+```
+(chroot)# ZPOOL_VDEV_NAME_PATH=1 grub-install /dev/sda
+Installing for i386-pc platform.
+Installation finished. No error reported.
+```
+
+Create and modify `/etc/dracut.conf.d/zol.conf` to configure Dracut to include
+at least the following values:
+
+```
+hostonly="yes"
+nofsck="yes"
+add_dracutmodules+=" zfs "
```

Comment:
Because the pool was created with disks out of `/dev/disk/by-id`, dracut also needs to know to look there. 
The following should also be added to `zol.conf`:

`persistent_policy="by-id"`

Reply:
Thanks for catching this